packages/flutter/lib/src/material/progress_indicator.dart

```diff
@@ -63,25 +65,29 @@ abstract class ProgressIndicator extends StatefulWidget {
 
   /// The progress indicator's background color.
   ///
-  /// The current theme's [ColorScheme.background] by default.
-  ///
-  /// This property is ignored if used in an adaptive constructor inside an iOS
-  /// environment.
+  /// It is up to the subclass to implement this in whatever way makes sense
+  /// for the given use case. See the subclass documentation for details.
   final Color? backgroundColor;
 
+  /// {@template flutter.progress_indicator.ProgressIndicator.color}
   /// The progress indicator's color.
   ///
-  /// This is only used if [valueColor] is null. If [color] is also null,
-  /// then it defaults to the current theme's [ColorScheme.primary] by default.
+  /// This is only used if [ProgressIndicator.valueColor] is null.
+  /// If [ProgressIndicator.color] is also null, then the ambient
+  /// [ProgressIndicatorThemeData.color] will be used. If that
+  /// is null then the current theme's [ColorScheme.primary] will
+  /// be used by default.
   ///
   /// This property is ignored if used in an adaptive constructor inside an iOS
   /// environment.
+  /// {@endtemplate}
   final Color? color;
 
   /// The progress indicator's color as an animated value.
   ///
-  /// If null, the progress indicator is rendered with [color], or if that is
-  /// also null then with the current theme's [ColorScheme.primary].
+  /// If null, the progress indicator is rendered with [color]. If that is null,
+  /// then it will use the ambient [ProgressIndicatorThemeData.color]. If that
+  /// is also null then it defaults to the current theme's [ColorScheme.primary].
   ///
   /// This property is ignored if used in an adaptive constructor inside an iOS
```

Comment:
As far as I can tell, this warning, which is all over the place, just refers to CircularProgressIndicator.adaptive.  Assuming that's true, we should just call out CircularProgressIndicator.adaptive specifically and eliminate this boilerplate comment where it's not really necessary. OK to do that in a separate PR BTW.